### PR TITLE
TELCODOCS-999  Support MultiNetworkPolicies in IPv6 networks

### PR DESCRIPTION
**Pull request by kquinn1204**
[TELCODOCS-999]: Support MultiNetworkPolicies in IPv6 networks

Version(s): 4.15, main

Issue:https://issues.redhat.com/browse/TELCODOCS-999

Link to docs preview: https://69719--ocpdocs-pr.netlify.app/openshift-enterprise/latest/networking/multiple_networks/configuring-multi-network-policy#nw-multi-network-policy-ipv6-support_configuring-multi-network-policy

QE review:
- [x] QE has approved this change.

Additional information:

### CODE REVIEW

**Comment by kquinn1204** [comment]
@zeeke Can you PTAL at this PR.

File: modules/nw-multi-network-policy-ipv6-suppport.adoc [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+    app: multi-networkpolicy
+data:
+  custom-v6-rules.txt: |
+    # accept NDP
+    -p icmpv6 --icmpv6-type neighbor-solicitation -j ACCEPT <1>
+    -p icmpv6 --icmpv6-type neighbor-advertisement -j ACCEPT <2>
+    # accept RA/RS
+    -p icmpv6 --icmpv6-type router-solicitation -j ACCEPT <3>
+    -p icmpv6 --icmpv6-type router-advertisement -j ACCEPT <4>
+    # accept redirect
```
Comment (zeeke) [inline modules/nw-multi-network-policy-ipv6-suppport.adoc:32]: At the moment, refirects  and packet-too-big packet types are not supported. 
https://github.com/openshift/cluster-network-operator/blob/master/bindata/network/multus-networkpolicy/custom-iptables-rules.yaml

- https://github.com/openshift/cluster-network-operator/pull/2010

File: modules/nw-multi-network-policy-ipv6-suppport.adoc [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+
+:_mod-docs-content-type: CONCEPT
+[id="nw-multi-network-policy-ipv6-support_{context}"]
+= Supporting multi-network policies in IPv6 networks
+
+The ICMPv6 Neighbor Discovery Protocol (NDP) plays a crucial role in IPv6 networks, facilitating the interaction between devices on the same link. NDP is a set of messages and processes that enable devices to discover and maintain information about neighboring nodes.
+
+The Cluster Network Operator (CNO) deploys the iptables implementation of multi-network policy when `useMultiNetworkPolicy` is set to `true`.
+
+To support multi-network policies in IPv6 networks the Cluster Network Operator deploys the set of rules defined in the following YAML.
```
Comment (zeeke) [inline modules/nw-multi-network-policy-ipv6-suppport.adoc:13]: I'm unsure if it's clear, but this predefined set of rules is deployed in every Pod affected by a MultiNetworkPolicy, not on the host.

**Review by zeeke — COMMENTED** [review]
Left a few comments

File: modules/nw-multi-network-policy-ipv6-suppport.adoc (hunk 2 above)
Comment (kquinn1204) [reply modules/nw-multi-network-policy-ipv6-suppport.adoc]: I think I made that clear now

Comment (zeeke) [reply modules/nw-multi-network-policy-ipv6-suppport.adoc]: Thanks!

**Comment by kquinn1204** [comment]
@kononovn Can you please review for QE.

File: modules/nw-multi-network-policy-ipv6-suppport.adoc [hunk 3]
```diff
@@ -0,0 +1,43 @@
+// Module included in the following assemblies:
+//
+// * networking/multiple_networks/configuring-multi-network-policy.adoc
+
+:_mod-docs-content-type: CONCEPT
+[id="nw-multi-network-policy-ipv6-support_{context}"]
+= Supporting multi-network policies in IPv6 networks
+
+The ICMPv6 Neighbor Discovery Protocol (NDP) plays a crucial role in IPv6 networks, facilitating the interaction between devices on the same link. NDP is a set of messages and processes that enable devices to discover and maintain information about neighboring nodes.
```
Comment (stevsmit) [inline modules/nw-multi-network-policy-ipv6-suppport.adoc:9]: There's no right or wrong here, but maybe defining what it is before defining its role is worth considering. Definitely up to you how you want to approach it. 

```suggestion
The ICMPv6 Neighbor Discovery Protocol (NDP) is a set of messages and processes that enable devices to discover and maintain information about neighboring nodes. NDP plays a crucial role in IPv6 networks, facilitating the interaction between devices on the same link. NDP 
```

File: modules/nw-multi-network-policy-ipv6-suppport.adoc [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+
+The Cluster Network Operator (CNO) deploys the iptables implementation of multi-network policy when `useMultiNetworkPolicy` is set to `true`.
```
Comment (stevsmit) [inline modules/nw-multi-network-policy-ipv6-suppport.adoc:11]:
```suggestion
The Cluster Network Operator (CNO) deploys the iptables implementation of multi-network policy when the `useMultiNetworkPolicy` field is set to `true`.
```

or "configuration field", or "parameter", etc. etc.

File: modules/nw-multi-network-policy-ipv6-suppport.adoc [hunk 5]
```diff
@@ -0,0 +1,43 @@
...
+
+To support multi-network policies in IPv6 networks the Cluster Network Operator deploys the following set of rules in every pod affected by a multi-network policy.
```
Comment (stevsmit) [inline modules/nw-multi-network-policy-ipv6-suppport.adoc:13]:
```suggestion
To support multi-network policies in IPv6 networks the Cluster Network Operator deploys the following set of rules in every pod affected by a multi-network policy:
```

File: modules/nw-multi-network-policy-ipv6-suppport.adoc [hunk 6]
```diff
@@ -0,0 +1,43 @@
...
+
```
Comment (stevsmit) [inline modules/nw-multi-network-policy-ipv6-suppport.adoc:14]: I wonder if this YAML example could benefit from a title?

.Example YAML name

That said, other YAML examples do not have names on this page, AFAICT.

File: modules/nw-multi-network-policy-ipv6-suppport.adoc [hunk 7]
```diff
@@ -0,0 +1,43 @@
...
+  custom-v6-rules.txt: |
+    # accept NDP
+    -p icmpv6 --icmpv6-type neighbor-solicitation -j ACCEPT <1>
+    -p icmpv6 --icmpv6-type neighbor-advertisement -j ACCEPT <2>
+    # accept RA/RS
+    -p icmpv6 --icmpv6-type router-solicitation -j ACCEPT <3>
+    -p icmpv6 --icmpv6-type router-advertisement -j ACCEPT <4>
+----
+
+<1> This rule allows incoming ICMPv6 Neighbor Solicitation messages, which are part of the neighbor discovery protocol (NDP). These messages help determine the link-layer addresses of neighboring nodes.
```
Comment (stevsmit) [inline modules/nw-multi-network-policy-ipv6-suppport.adoc:33]: In the first paragraph, you use title case (the first letter of each word in Neighbor Discovery Protocol is capitalized). Here, you do not (they're all lower case). Not sure which it should be, but we should be consistent.

Comment (kquinn1204) [reply modules/nw-multi-network-policy-ipv6-suppport.adoc:33]: going with lower case to be consistent

**Comment by kquinn1204** [comment]
@kononovn Can you please review for QE.

**Review by kononovn — APPROVED** [review]
/lgtm

**Pull request merged**